docs/templates/tutorials/subclass-tuner.md

```diff
@@ -0,0 +1,132 @@
+# Subclassing Tuner for Custom Training Loops
+
+The `Tuner` class at `kerastuner.engine.tuner.Tuner` can be subclassed to support advanced uses such as:
+
+- Custom training loops (GANs, reinforement learning, etc.)
+- Adding hyperparameters outside of the model builing function (preprocessing, data augmentation, test time augmentation, etc.)
+
+This tutorial will not cover subclassing to support non-Keras models. To accomplish this, you can subclass the `kerastuner.engine.base_tuner.BaseTuner` class (See `kerastuner.tuners.sklearn.Sklearn` for an example).
+
+### Understanding the search process.
+
+`Tuner.search` can be passed any arguments. These arguments will be passed directly to `Tuner.run_trial`, along with a `Trial` object that contains information about the current trial, including hyperparameters and the status of the trial. Typically, `Tuner.run_trial` is the only method that users need to override when subclassing `Tuner`.
+
+### Example
+
+```python
```

Comment:
This is a complex example. We should consider using a format that enables us to make the example directly runnable in Colab in one click, and easily testable in an automatically generated integration test.

For instance, examples on tensorflow.org are typically shipped as runnable notebooks. The notebooks are used to generated a HTML webpage (it goes through markdown first), can be launched in Colab, and are run as part of integration tests.

Reply:
Is there a way to do the equivalent on GH? I couldn't find anything from googling around a bit

I ran this example on my computer locally to make sure it worked, is that ok for now?

Reply:
There is no ready-made solution that I'm aware of, we'd have to roll out our own (and make it shared cross all of our repos). We'll do this eventually, but we can't block tutorials based on this, so let's ship the tutorial as is for now.

Reply:
Sounds good, made a tracking bug here: https://github.com/keras-team/keras-autodoc/issues/35